segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -114,6 +117,8 @@ public PravegaConnectionListener(boolean ssl, String host, int port, StreamSegme
             this.tokenVerifier = new PassingTokenVerifier();
         }
         this.replyWithStackTraceOnError = replyWithStackTraceOnError;
+        this.executor = ExecutorServiceHelpers.newScheduledThreadPool(Runtime.getRuntime().availableProcessors(),
```

Comment:
Please do not create yet another executor service. I have a PR out that will consolidate some of these executors into less, so that we don't end up having too many idle threads sitting around.

Reply:
I saw where you used this. It's because of `Futures.loop`.

I wonder if we can fix this by adding this to `AsyncIterator.java`:
```
 default CompletableFuture<Void> forEachRemaining(Predicate<? super T> consumer) {
        return getNext().thenCompose(e -> {
            boolean canContinue = e != null && consumer.test(e);
            if (canContinue) {
                return forEachRemaining(consumer);
            } else {
                return CompletableFuture.completedFuture(null);
            }
        });
    }
```

That would eliminate the need for an executor and you can replace your entire loop in the methods below with `.forEachRemaining(Predicate)` and return `false` from your `Predicate` callback when you want to stop the loop.

Reply:
I do not see how the usage of an executor can be eliminated as `forEachRemaining( consumer, executor)` would still require an executor 🤔 
Could you please provide additional info?

Reply:
That is not what I was suggesting. I asked you to add another method to `AsyncIterator` which does not require an Executor, and you can use that to replace your loop.

Reply:
done.